Code/Framework/AzCore/AzCore/Math/Vector4.inl

```diff
@@ -28,21 +31,42 @@ namespace AZ
         ;
     }
 
-
     AZ_MATH_INLINE Vector4::Vector4(Simd::Vec4::FloatArgType value)
         : m_value(value)
     {
-        ;
     }
 
     AZ_MATH_INLINE Vector4::Vector4(const Vector2& source)
+    : m_value(Simd::Vec4::FromVec2(source.GetSimdValue()))
     {
-        *this = CreateFromVector3(Vector3(source));
+        m_z = 0.0f;
+        m_w = 1.0f;
+    }
+
+    AZ_MATH_INLINE Vector4::Vector4(const Vector2& source, float z) 
+    : m_value(Simd::Vec4::FromVec2(source.GetSimdValue()))
+    {
+        m_z = z;
+        m_w = 1.0f;
+    }
+
+    AZ_MATH_INLINE Vector4::Vector4(const Vector2& source, float z, float w) 
+    : m_value(Simd::Vec4::FromVec2(source.GetSimdValue()))
+    {
+       m_z = z;
+       m_w = w;
     }
 
     AZ_MATH_INLINE Vector4::Vector4(const Vector3& source)
+    : m_value(Simd::Vec4::FromVec3(source.GetSimdValue()))
+    {
+        m_w = 1.0f;
+    }
+
+    AZ_MATH_INLINE Vector4::Vector4(const Vector3& source, float w) 
+    : m_value(Simd::Vec4::FromVec3(source.GetSimdValue()))
```

Comment:
nit: Formatting looks a bit off here, need the 4 space indent for the initializer list